File: src/test/java/com/medicalsystem/repository/ApplicationUserRepositoryTest.java
package com.medicalsystem.repository;

import com.medicalsystem.model.ApplicationUser;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

@RunWith(SpringRunner.class)
@SpringBootTest
public class ApplicationUserRepositoryTest {

    @Autowired
    private ApplicationUserRepository applicationUserRepository;

    private long id;

    @Test
    public void test_autowire() {
        assertNotNull(applicationUserRepository);
    }

    @Before
    public void setup() {
        ApplicationUser user = new ApplicationUser();
        user.setUsername("mikolaj");
        user = applicationUserRepository.save(user);
        id = user.getId();
    }

    @After
    public void tearDown() {
        applicationUserRepository.delete(id);
    }

    @Test
    public void test_existsByUsername() {
        assertTrue(applicationUserRepository.existsByUsername("mikolaj"));
    }

    @Test
    public void test_findByUsername() {
        ApplicationUser user = applicationUserRepository.findByUsername("mikolaj");
        assertEquals(id, user.getId());
    }
}

File: src/test/kotlin/com/medicalsystem/repository/ApplicationUserRepositoryTest.kt
package com.medicalsystem.repository

import com.medicalsystem.model.ApplicationUser
import org.junit.After
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.dao.DataIntegrityViolationException
import org.springframework.test.context.junit4.SpringRunner
import kotlin.test.assertFailsWith

@RunWith(SpringRunner::class)
@SpringBootTest
class ApplicationUserRepositoryTest {

    @Autowired
    lateinit var applicationUserRepository: ApplicationUserRepository

    @Before @After
    fun tearDown() = applicationUserRepository.deleteAll()

    @Test
    fun testAutowire() {
        Assert.assertNotNull(applicationUserRepository)
    }

    @Test
    fun testSave() {
        val user = ApplicationUser(username = "msieniawski", password = "pw123")
        val savedUser = applicationUserRepository.save(user)
        Assert.assertTrue(applicationUserRepository.exists(savedUser.id))
    }

    @Test
    fun testPersistSameUsernamesShouldThrowAnException() {
        val user1 = ApplicationUser(username = "msieniawski", password = "pw123")
        val user2 = ApplicationUser(username = "msieniawski", password = "pw123")

        applicationUserRepository.save(user1)

        assertFailsWith(DataIntegrityViolationException::class) {
            applicationUserRepository.save(user2)
        }
    }
}